clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -307,9 +315,11 @@ public int hashCode() {
         for (int i = 0; i < this.values.length; i++) {
             Field f = this.schema.get(i);
             if (f.type() instanceof ArrayOf) {
-                Object[] arrayObject = (Object []) this.get(f);
-                for (Object arrayItem: arrayObject)
-                    result = prime * result + arrayItem.hashCode();
+                if (this.get(f) != null) {
```

Comment:
Not sure if it matters, but we will get the exact same hash for empty and null arrays?